### PR TITLE
fix: Avoid pipeline status update for API deployments

### PR DESCRIPTION
**Pull request by chandrasekharan-zipstack**
## What

- Suppress exceptions that occur when pipeline status is updated
- MINOR: Fixed error handling logic in prompt service  

## Why

- API dpeloyments don't maintain the last run status hence any updates here causes an issue since there's no entry for it in `Pipeline`

## How

-

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- No

## Related Issues or PRs

- #381 

## Notes on Testing

- Able to obtain response for prev deployed APIs and new ones too

## Screenshots
image

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Review by jagadeeswaran-zipstack — APPROVED** [review]

File: backend/pdm.lock [hunk 1]
```diff
@@ -5,7 +5,7 @@
 groups = ["default", "deploy", "test"]
 strategy = ["cross_platform", "inherit_metadata"]
 lock_version = "4.4.1"
-content_hash = "sha256:65995bec9035d980c5dbfa609f531964e573eb0bb13fa53c207a4ae006c253b0"
+content_hash = "sha256:5924a0dcf9f3a5a94264fa1ed7a3d8bd5a2c7db3b565a2ba887c93e06dbb0e09"
```
Comment (muhammad-ali-e) [inline backend/pdm.lock:8]: Why it updated?

**Review by muhammad-ali-e — APPROVED** [review]
LGTM

File: backend/pdm.lock (hunk 1 above)
Comment (chandrasekharan-zipstack) [reply backend/pdm.lock:8]: I think it must have been missed in `main`

File: backend/workflow_manager/workflow/workflow_helper.py [hunk 2]
```diff
@@ -267,35 +267,53 @@ def run_workflow(
...
-                log_id=str(execution_service.execution_log_id),
-                error=updated_execution.error_message,
-                mode=updated_execution.execution_mode,
-                result=destination.api_results,
+        # Expected exception since API deployments are not tracked in Pipeline
+        except Pipeline.DoesNotExist:
+            pass
+        except Exception as e:
+            logger.warning(
+                f"Error updating pipeline {pipeline_id} status: {e} "
```
Comment (muhammad-ali-e) [inline backend/workflow_manager/workflow/workflow_helper.py:314]:
```suggestion
                f"Error updating pipeline {pipeline_id} status: {e}, "
```

**Review by muhammad-ali-e — APPROVED** [review]
LGTM

**Pull request merged**